src/Altinn.App.Core/Features/Correspondence/Builder/ICorrespondenceNotificationBuilder.cs

```diff
@@ -0,0 +1,115 @@
+using Altinn.App.Core.Features.Correspondence.Models;
+
+namespace Altinn.App.Core.Features.Correspondence.Builder;
+
+/// <summary>
+/// Indicates that the <see cref="CorrespondenceNotificationBuilder"/> instance is on the <see cref="CorrespondenceNotification.NotificationTemplate"/> step
+/// </summary>
+public interface ICorrespondenceNotificationBuilderTemplate
+{
+    /// <summary>
+    /// Sets the notification template for the correspondence notification
+    /// </summary>
+    /// <param name="notificationTemplate">The notification template</param>
+    ICorrespondenceNotificationBuilder WithNotificationTemplate(
+        CorrespondenceNotificationTemplate notificationTemplate
+    );
+}
+
+/// <summary>
+/// Indicates that the <see cref="CorrespondenceNotificationBuilder"/> instance has completed all required steps and can proceed to <see cref="CorrespondenceNotificationBuilder.Build"/>
+/// </summary>
+public interface ICorrespondenceNotificationBuilder : ICorrespondenceNotificationBuilderTemplate
```

Comment:
One thing I came to think of now, a bit late in the game, but are we using the interfaces for something? I.e. could the builder all just be records that don't implement interfaces? Reason I ask is because usually we have interfaces because we want to hide implementation details and have the expected behavior defined independently. In this case the consumers are using the classes directly

Reply:
Discussed offline, reason being we have some guardrails in the form of interfaces to ensure required fields are input